### ISSUE TITLE
WebView and PlatformView causes soft reboot on Android Q due to exhaustion of resources

### ISSUE THREAD

**Issue by brsun** [flutter/flutter#54265]
Hi, we have a product use flutter, recently I receive some error reports from our end users about WebView crash.  so, I start to investigate the problem. 

I created a simple project like this https://github.com/brsun/webview_crash_test 
It has two pages, click the float button on the first page will navigate to the second WebView page and create a WebView, then click back will return to the first page and dispose WebView.  
Then I run an automated test to repeat this process. It always crash and lead the system to soft reboot on Android 10 after repeat the test case hundreds of times.

I tested this case on several phones with different models. The crash only occurs on Android 10.
Though I got different logs from different phone, some are registered too many receivers, some are create too many surface, It seems that the WebView does not release system resource correctly.I even doubt it relate to the multi-instance of input method feature.

The resource will not release even after kill the app process, and will be accumulated as long as system running.  Though the crashing need to create WebView hundreds of times, it seems easier to be reproduced by user than I thought.

## Steps to Reproduce

1. Create a project with two pages, the second page has a WebView. Or clone https://github.com/brsun/webview_crash_test
2. Navigate to the second page which has WebView
3. Back to the first page and dispose WebView
4. Repeat hundreds of times

**Expected results:** Running correctly

**Actual results:** The system reboot

<details>
  <summary>Logs</summary>

Env: XiaoMi MIX 3, Android Q: 
```04-07 21:14:28.982  1619  1726 I DisplayManagerService: Display DisplayDeviceInfo{"flutter-vd": uniqueId="virtual:com.accumulus.flutterapp,10227,flutter-vd,0", 1080 x 1960, modeId 743, defaultModeId 743, supportedModes [{id=743, width=1080, height=1960, fps=60.0}], colorMode 0, supportedColorModes [0], HdrCapabilities null, density 440, 440.0 x 440.0 dpi, appVsyncOff 0, presDeadline 16666666, touch NONE, rotation 0, type VIRTUAL, state ON, owner com.accumulus.flutterapp (uid 10227), FLAG_PRIVATE, FLAG_NEVER_BLANK, FLAG_OWN_CONTENT_ONLY} does not support input device matching.
04-07 21:14:28.986  1619 32564 E AndroidRuntime: *** FATAL EXCEPTION IN SYSTEM PROCESS: Thread-771
04-07 21:14:28.986  1619 32564 E AndroidRuntime: java.lang.IllegalStateException: Too many receivers, total of 1000, registered for pid: 1619, callerPackage: android
04-07 21:14:28.986  1619 32564 E AndroidRuntime: 	at com.android.server.am.ActivityManagerService.registerReceiver(ActivityManagerService.java:14846)
04-07 21:14:28.986  1619 32564 E AndroidRuntime: 	at android.app.ContextImpl.registerReceiverInternal(ContextImpl.java:1537)
04-07 21:14:28.986  1619 32564 E AndroidRuntime: 	at android.app.ContextImpl.registerReceiver(ContextImpl.java:1498)
04-07 21:14:28.986  1619 32564 E AndroidRuntime: 	at com.android.server.wm.DisplayRotation$1.run(DisplayRotation.java:195)
```

Env: OnePlus 7, Android Q
```
04-07 20:32:05.908 22865 25111 I DisplayManagerService: Display device added: DisplayDeviceInfo{"flutter-vd": uniqueId="virtual:com.accumulus.flutterapp,10264,flutter-vd,0", 1080 x 1987, modeId 702, defaultModeId 702, supportedModes [{id=702, width=1080, height=1987, fps=60.0}], colorMode 0, supportedColorModes [0], HdrCapabilities null, density 420, 420.0 x 420.0 dpi, appVsyncOff 0, presDeadline 16666666, touch NONE, rotation 0, type VIRTUAL, state ON, owner com.accumulus.flutterapp (uid 10264), FLAG_PRIVATE, FLAG_NEVER_BLANK, FLAG_OWN_CONTENT_ONLY} ownerUid:10264
04-07 20:32:05.908 22865 25111 D OPBF    : UidStateCache makeQuickJudgment, uid:10264 to do nothing for totalStateMap1:9814 |needToMeetMap:0 |freezeState:false |reason:check virtualDeviceChange add:true uniqueId:virtual:com.accumulus.flutterapp,10264,flutter-vd,0
04-07 20:32:05.917 22865 22894 D SystemGestures: mSwipeStartThreshold=112 mDisplayCutoutTouchableRegionSize=32 mSwipeDistanceThreshold=112
04-07 20:32:05.921 22865 23207 D OpQuickReply: updateIMEMarginBottom force: true
04-07 20:32:05.922 22865 30449 E AndroidRuntime: *** FATAL EXCEPTION IN SYSTEM PROCESS: Thread-934
04-07 20:32:05.922 22865 30449 E AndroidRuntime: java.lang.IllegalStateException: Too many receivers, total of 1000, registered for pid: 22865, callerPackage: android
04-07 20:32:05.922 22865 30449 E AndroidRuntime: 	at com.android.server.am.ActivityManagerService.registerReceiver(ActivityManagerService.java:15745)
04-07 20:32:05.922 22865 30449 E AndroidRuntime: 	at android.app.ContextImpl.registerReceiverInternal(ContextImpl.java:1561)
04-07 20:32:05.922 22865 30449 E AndroidRuntime: 	at android.app.ContextImpl.registerReceiver(ContextImpl.java:1512)
04-07 20:32:05.922 22865 30449 E AndroidRuntime: 	at com.android.server.wm.DisplayRotation$2.run(DisplayRotation.java:218)
```

Env HUAWEI P30 Pro, Android Q
```
04-06 18:46:55.961 27003 27003 F chromium: [FATAL:crashpad_client_linux.cc(405)] Render process (27431)'s crash wasn't handled by all associated  webviews, triggering application crash.
04-06 18:46:55.962 27003 27003 F libc    : Fatal signal 5 (SIGTRAP), code 1 (TRAP_BRKPT), fault addr 0x73ae55e7d0 in tid 27003 (mulus.assistant), pid 27003 (mulus.assistant)
```

Env HUAWEI P30 Pro, Android Q
```
04-03 16:59:33.306   719   815 E SurfaceFlinger: AddClientLayer failed, mNumLayers (4097) >= MAX_LAYERS (4096)
04-03 16:59:33.307 13350 13382 E SurfaceComposerClient: SurfaceComposerClient::createSurface error Out of memory
04-03 16:59:33.308 13350 13382 W WindowManager: Unable to allocate black surface
04-03 16:59:33.308 13350 13382 W WindowManager: android.view.Surface$OutOfResourcesException
04-03 16:59:33.308 13350 13382 W WindowManager: 	at android.view.SurfaceControl.nativeCreate(libmapleframework.so:7468004)
04-03 16:59:33.308 13350 13382 W WindowManager: 	at android.view.SurfaceControl.<init>(libmapleframework.so:7468848)
04-03 16:59:33.308 13350 13382 W WindowManager: 	at android.view.SurfaceControl.<init>(libmapleframework.so:7469208)
04-03 16:59:33.308 13350 13382 W WindowManager: 	at android.view.SurfaceControl$Builder.build(libmapleframework.so:7469508)
04-03 16:59:33.308 13350 13382 W WindowManager: 	at com.android.server.wm.BlackFrame$BlackSurface.<init>(libmapleservices.so:7965872)
04-03 16:59:33.308 13350 13382 W WindowManager: 	at com.android.server.wm.RootWindowContainer.performSurfacePlacementNoTrace(libmapleservices.so:6951296)
04-03 16:59:33.308 13350 13382 W WindowManager: 	at com.android.server.wm.RootWindowContainer.performSurfacePlacement(libmapleservices.so:6953828)
04-03 16:59:33.308 13350 13382 W WindowManager: 	at com.android.server.wm.WindowSurfacePlacer.performSurfacePlacementLoop(libmapleservices.so:6960896)
04-03 16:59:33.308 13350 13382 W WindowManager: 	at com.android.server.wm.WindowSurfacePlacer.performSurfacePlacement(libmapleservices.so:6943900)
```
</details>

<details>
  <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.3 19D76, locale zh-Hans-CN)
    • Flutter version 1.12.13+hotfix.8 at /Users/brian/development/flutter
    • Framework revision 0b8abb4724 (8 周前), 2020-02-11 11:44:36 -0800
    • Engine revision e1e6ced81d
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/brian/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4, Build version 11E146
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.43.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.0

[!] Connected device
    ! No devices available
```
</detail>

**Comment by CodePoem** [comment]
I had the same problem.When will it be repaired？

**Comment by pedromassangocode** [comment]
> Then I run an automated test to repeat this process. It always crash and lead the system to soft reboot on Android 10 after repeat the test case hundreds of times.

Hi @brsun 
Can you provide the steps to reproduce the issue? As you know it is hard to reproduce this manually so I need you to provide the steps you used to automate this process.
Thank you

**Comment by brsun** [comment]
Oh, I almost forget it.
I haven't reproduce this issue on the release version. 
Maybe fixed by webview_flutter  version 1.0.0+，with flutter version 1.22.

**Comment by TahaTesser** [comment]
Hi @brsun 
Given your last message I feel safe to close this issue, if you disagree/ experience the issue again please write in the comments and I will reopen it. 
Thank you
